### PR TITLE
add(letters): Add letters argument as per Dustin Kirklands implementation

### PR DESCRIPTION
**Pull request by robbash**
Hey @anthonkendel. Found your JS implementation today and much appreciate the work.

One thing I was missing so far is the `--letters` argument (limitation of the word length) hence why I created this change.

### CODE REVIEW

File: src/petname.js [hunk 1]
```diff
@@ -14,14 +14,26 @@ function getPart(parts) {
...
  * @returns {string}
  */
-function generatePetname({ words = 2, separator = '-' } = {}) {
+function generatePetname({ words = 2, separator = '-', letters = 0 } = {}) {
   let petnameParts = [];
 
+  if (letters > 0) {
+    const filterFunc = filterByLength(letters);
+
+    adjectives = adjectives.filter(filterFunc);
```
Comment (anthonkendel) [inline src/petname.js]: inline the filters, and remove the filterByLength function to keep it simpler.

```
adjectives.filter(v => v.length <= letters);
adverbs.filter(v => v.length <= letters);
names.filter(v => v.length <= letters);
```

File: src/petname.js [hunk 2]
```diff
@@ -14,14 +14,26 @@ function getPart(parts) {
...
+}
+
 /**
- * @param {{ words: number, separator: string }} parameters
+ * @param {{ words: number, separator: string, letters: number }} parameters
  *
  * @returns {string}
  */
-function generatePetname({ words = 2, separator = '-' } = {}) {
+function generatePetname({ words = 2, separator = '-', letters = 0 } = {}) {
```
Comment (anthonkendel) [inline src/petname.js]: use `undefined` as default value for `letters`

**Review by anthonkendel — CHANGES_REQUESTED** [review]
Mostly looks fine, some minor comments.

**Comment by robbash** [comment]
Thanks. Code updated...

**Review by anthonkendel — APPROVED** [review]

**Pull request merged**